core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala

```diff
@@ -171,16 +171,16 @@ private[spark] object ResourceUtils extends Logging {
   // integer amount and the number of slots per address. For instance, if the amount is 0.5,
```

Comment:
there are comments in function getSchedulerTaskResourceAmount that need to change based  on these new changes. Its also unclear to me from your response whether that function and warnOnWastedResources are still used

Reply:
Yes, `getSchedulerTaskResourceAmount` is only used by warnOnWastedResources to calculate the max task number per executor, please refer to [here](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/resource/ResourceUtils.scala#L474-L476), while warnOnWastedResources will be used when checking resource profile, or calculateTasksAndLimitingResource, Please refer to [here](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/resource/ResourceProfile.scala#L228) and [here](https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/SparkContext.scala#L3155)